### PR TITLE
[tune] Clean up result logging: move out of /tmp, add timestamp

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

Clean up result logging: move out of /tmp, add timestamp, validation accuracy field.

https://github.com/ray-project/ray/issues/995
https://github.com/ray-project/ray/issues/1300

### CODE REVIEW

File: python/ray/rllib/agent.py [hunk 1]
```diff
@@ -82,12 +81,15 @@ def __init__(
...
-            if not os.path.exists(self._default_logdir):
-                os.makedirs(self._default_logdir)
+            if not os.path.exists(DEFAULT_RESULTS_DIR):
+                os.makedirs(DEFAULT_RESULTS_DIR)
             self.logdir = tempfile.mkdtemp(
-                prefix=logdir_suffix, dir=self._default_logdir)
+                prefix=logdir_suffix, dir=DEFAULT_RESULTS_DIR)
             self._result_logger = UnifiedLogger(self.config, self.logdir, None)
 
+        # Also set the logdir as the pwd for user-defined loggers.
```
Comment (richardliaw) [inline python/ray/rllib/agent.py]: This is not very visible - would it be reasonable to move this to docstring? Also the "for user-defined loggers" part is a little confusing - would be ok with removing it.

**Review by richardliaw — APPROVED** [review]
one comment else lgtm

File: python/ray/rllib/agent.py (hunk 1 above)
Comment (ericl) [reply python/ray/rllib/agent.py]: On second thought, I realized chdir is actually a problem when running agents in the same process. I moved this code to the trial class, but I guess in general there's not a great solution here.

Comment (richardliaw) [reply python/ray/rllib/agent.py]: this is fine; will merge after tests

**Pull request merged**